ICE: index out of bounds: the len is 1 but the index is 1 -Cinstrument-coverage

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/

-->

### Code
found via fuzzing
```Rust
#![no_core]
#![feature(no_core, lang_items)]
#[lang="sized"]
trait Sized { }

fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    let x = x << 1;
}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.66.0-nightly (c0983a9aa 2022-10-12)
binary: rustc
commit-hash: c0983a9aac889d16722a12602ac678051e62c3fb
commit-date: 2022-10-12
host: x86_64-unknown-linux-gnu
release: 1.66.0-nightly
LLVM version: 15.0.2
```

### Error output
`rustc -Cinstrument-coverage file.rs`
```
error: `self` parameter is only allowed in associated functions
 --> icemaker_87671.rs:6:15
  |
6 | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
  |               ^^^^ not semantically valid as function parameter
  |
  = note: associated functions are those in `impl` or `trait` definitions

error[E0412]: cannot find type `Struct` in this scope
 --> icemaker_87671.rs:6:22
  |
6 | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
  |                      ^^^^^^ not found in this scope

error[E0425]: cannot find value `x` in this scope
 --> icemaker_87671.rs:7:13
  |
7 |     let x = x << 1;
  |             ^ help: a local variable with a similar name exists: `f`

error: `profiler_builtins` crate (required by compiler options) is not compatible with crate attribute `#![no_core]`

error[E0601]: `main` function not found in crate `icemaker_87671`
 --> icemaker_87671.rs:8:2
  |
8 | }
  |  ^ consider adding a `main` function to `icemaker_87671.rs`

error[E0152]: found duplicate lang item `sized`
 --> icemaker_87671.rs:4:1
  |
4 | trait Sized { }
  | ^^^^^^^^^^^
  |
  = note: the lang item is first defined in crate `core`.
  = note: first definition in `core` loaded from /home/matthias/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-05898138a596088a.rlib
  = note: second definition in the local crate (`icemaker_87671`)

error: `#[panic_handler]` function required, but not found

error: language item required, but not found: `eh_personality`
  |
  = note: this can occur when a binary crate with `#![no_std]` is compiled for a target where `eh_personality` is defined in the standard library
  = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```

thread 'rustc' panicked at 'index out of bounds: the len is 1 but the index is 1', /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/compiler/rustc_middle/src/ty/subst.rs:371:43
stack backtrace:
   0:     0x7f2bca6995b0 - std::backtrace_rs::backtrace::libunwind::trace::h0bf6f02637fe6971
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x7f2bca6995b0 - std::backtrace_rs::backtrace::trace_unsynchronized::hca573de4ebdee142
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f2bca6995b0 - std::sys_common::backtrace::_print_fmt::h78a3b0f3fa5431ce
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7f2bca6995b0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h388012888db78658
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7f2bca6f559e - core::fmt::write::h8541eee12bfb22e1
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/core/src/fmt/mod.rs:1209:17
   5:     0x7f2bca689835 - std::io::Write::write_fmt::haf5892ef07750179
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/io/mod.rs:1680:15
   6:     0x7f2bca69c353 - std::sys_common::backtrace::_print::hd2a58ab324867b58
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7f2bca69c353 - std::sys_common::backtrace::print::h62650e0a0854a421
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7f2bca69c353 - std::panicking::default_hook::{{closure}}::h0c0249b3670af819
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/panicking.rs:267:22
   9:     0x7f2bca69c02a - std::panicking::default_hook::hdd3bdd204a1e5f34
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/panicking.rs:286:9
  10:     0x7f2bccfd8641 - rustc_driver[1f63e0707e3aafaf]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7f2bca69cb8b - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h6f493fd9c4eb244a
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/alloc/src/boxed.rs:1952:9
  12:     0x7f2bca69cb8b - std::panicking::rust_panic_with_hook::hcdeb5a66936a633b
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/panicking.rs:673:13
  13:     0x7f2bca69c9e7 - std::panicking::begin_panic_handler::{{closure}}::h76d64f438650eafd
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/panicking.rs:560:13
  14:     0x7f2bca699a5c - std::sys_common::backtrace::__rust_end_short_backtrace::h6159ba2d4e07f4de
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys_common/backtrace.rs:138:18
  15:     0x7f2bca69c702 - rust_begin_unwind
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/panicking.rs:556:5
  16:     0x7f2bca6f2173 - core::panicking::panic_fmt::h811742b56cf0125a
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/core/src/panicking.rs:142:14
  17:     0x7f2bca6f20b2 - core::panicking::panic_bounds_check::hfee18f8513cbc676
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/core/src/panicking.rs:84:5
  18:     0x7f2bcde70c44 - <rustc_infer[47b922782ea47410]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[a031dbb00953ed73]::traits::error_reporting::suggestions::TypeErrCtxtExt>::suggest_floating_point_literal
  19:     0x7f2bcde75abd - <rustc_infer[47b922782ea47410]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[a031dbb00953ed73]::traits::error_reporting::TypeErrCtxtExt>::report_selection_error
  20:     0x7f2bcde7dc4a - <rustc_infer[47b922782ea47410]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[a031dbb00953ed73]::traits::error_reporting::InferCtxtPrivExt>::report_fulfillment_error
  21:     0x7f2bcde71cf2 - <rustc_infer[47b922782ea47410]::infer::error_reporting::TypeErrCtxt as rustc_trait_selection[a031dbb00953ed73]::traits::error_reporting::TypeErrCtxtExt>::report_fulfillment_errors
  22:     0x7f2bcbc50540 - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_overloaded_binop
  23:     0x7f2bcbbc36be - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_binop
  24:     0x7f2bcbbb6213 - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  25:     0x7f2bcbbdf451 - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_decl
  26:     0x7f2bcbbdbaeb - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_block_with_expected
  27:     0x7f2bcbbb4bfe - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  28:     0x7f2bcc50fba6 - <rustc_hir_analysis[a6a5d63f9823a7da]::check::fn_ctxt::FnCtxt>::check_return_expr
  29:     0x7f2bcc506850 - rustc_hir_analysis[a6a5d63f9823a7da]::check::check::check_fn
  30:     0x7f2bcc4f813c - <rustc_hir_analysis[a6a5d63f9823a7da]::check::inherited::InheritedBuilder>::enter::<rustc_hir_analysis[a6a5d63f9823a7da]::check::typeck_with_fallback<rustc_hir_analysis[a6a5d63f9823a7da]::check::typeck::{closure#0}>::{closure#1}, &rustc_middle[a8ed4a0dc47aac3]::ty::context::TypeckResults>
  31:     0x7f2bcc4f700c - rustc_hir_analysis[a6a5d63f9823a7da]::check::typeck
  32:     0x7f2bcc617595 - rustc_query_system[a76597d842537d9]::query::plumbing::try_execute_query::<rustc_query_impl[1e7ed5a7eb258aab]::plumbing::QueryCtxt, rustc_query_system[a76597d842537d9]::query::caches::DefaultCache<rustc_span[ddec628aeed15f05]::def_id::LocalDefId, &rustc_middle[a8ed4a0dc47aac3]::ty::context::TypeckResults>>
  33:     0x7f2bcced91ee - <rustc_query_impl[1e7ed5a7eb258aab]::Queries as rustc_middle[a8ed4a0dc47aac3]::ty::query::QueryEngine>::typeck
  34:     0x7f2bccd09df7 - rustc_data_structures[fec937313883fa6d]::sync::par_for_each_in::<&[rustc_span[ddec628aeed15f05]::def_id::LocalDefId], <rustc_middle[a8ed4a0dc47aac3]::hir::map::Map>::par_body_owners<rustc_hir_analysis[a6a5d63f9823a7da]::check::typeck_item_bodies::{closure#0}>::{closure#0}>
  35:     0x7f2bccd09bb3 - rustc_hir_analysis[a6a5d63f9823a7da]::check::typeck_item_bodies
  36:     0x7f2bccc2b66a - rustc_query_system[a76597d842537d9]::query::plumbing::try_execute_query::<rustc_query_impl[1e7ed5a7eb258aab]::plumbing::QueryCtxt, rustc_query_system[a76597d842537d9]::query::caches::DefaultCache<(), ()>>
  37:     0x7f2bccc2b397 - rustc_query_system[a76597d842537d9]::query::plumbing::get_query::<rustc_query_impl[1e7ed5a7eb258aab]::queries::typeck_item_bodies, rustc_query_impl[1e7ed5a7eb258aab]::plumbing::QueryCtxt>
  38:     0x7f2bcc9d07cf - <rustc_session[8b98ba4aacc9d309]::session::Session>::time::<(), rustc_hir_analysis[a6a5d63f9823a7da]::check_crate::{closure#7}>
  39:     0x7f2bcc9d039f - rustc_hir_analysis[a6a5d63f9823a7da]::check_crate
  40:     0x7f2bcc9cfea7 - rustc_interface[cf2fa28a67e8bef]::passes::analysis
  41:     0x7f2bccd629e4 - rustc_query_system[a76597d842537d9]::query::plumbing::try_execute_query::<rustc_query_impl[1e7ed5a7eb258aab]::plumbing::QueryCtxt, rustc_query_system[a76597d842537d9]::query::caches::DefaultCache<(), core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>>
  42:     0x7f2bccd62717 - rustc_query_system[a76597d842537d9]::query::plumbing::get_query::<rustc_query_impl[1e7ed5a7eb258aab]::queries::analysis, rustc_query_impl[1e7ed5a7eb258aab]::plumbing::QueryCtxt>
  43:     0x7f2bcbc91044 - <rustc_interface[cf2fa28a67e8bef]::passes::QueryContext>::enter::<rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}::{closure#2}::{closure#3}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>
  44:     0x7f2bcbc8d496 - <rustc_interface[cf2fa28a67e8bef]::interface::Compiler>::enter::<rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}::{closure#2}, core[fae631be8e3c81b5]::result::Result<core[fae631be8e3c81b5]::option::Option<rustc_interface[cf2fa28a67e8bef]::queries::Linker>, rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>
  45:     0x7f2bcbc8843c - rustc_span[ddec628aeed15f05]::with_source_map::<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_interface[cf2fa28a67e8bef]::interface::create_compiler_and_run<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}>::{closure#1}>
  46:     0x7f2bcbc87e3f - rustc_interface[cf2fa28a67e8bef]::interface::create_compiler_and_run::<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}>
  47:     0x7f2bcbc87032 - <scoped_tls[4da31d6f3f5146b7]::ScopedKey<rustc_span[ddec628aeed15f05]::SessionGlobals>>::set::<rustc_interface[cf2fa28a67e8bef]::interface::run_compiler<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}>::{closure#0}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>
  48:     0x7f2bcbc8650f - std[7bd0f41fe60b70b6]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[cf2fa28a67e8bef]::util::run_in_thread_pool_with_globals<rustc_interface[cf2fa28a67e8bef]::interface::run_compiler<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}>::{closure#0}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>::{closure#0}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>
  49:     0x7f2bcbc8637f - <<std[7bd0f41fe60b70b6]::thread::Builder>::spawn_unchecked_<rustc_interface[cf2fa28a67e8bef]::util::run_in_thread_pool_with_globals<rustc_interface[cf2fa28a67e8bef]::interface::run_compiler<core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>, rustc_driver[1f63e0707e3aafaf]::run_compiler::{closure#1}>::{closure#0}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>::{closure#0}, core[fae631be8e3c81b5]::result::Result<(), rustc_errors[d5334a4a25fbb394]::ErrorGuaranteed>>::{closure#1} as core[fae631be8e3c81b5]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  50:     0x7f2bca6a6793 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hc67cb2c032770d7f
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/alloc/src/boxed.rs:1938:9
  51:     0x7f2bca6a6793 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::he79e59a805a8533c
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/alloc/src/boxed.rs:1938:9
  52:     0x7f2bca6a6793 - std::sys::unix::thread::Thread::new::thread_start::hae509166483801f9
                               at /rustc/db0597f5619d5ed93feca28e61226d3581cc7867/library/std/src/sys/unix/thread.rs:108:17
  53:     0x7f2bca3fc8fd - <unknown>
  54:     0x7f2bca47ea60 - <unknown>
  55:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.66.0-nightly (db0597f56 2022-10-11) running on x86_64-unknown-linux-gnu

note: compiler flags: -C instrument-coverage

query stack during panic:
#0 [typeck] type-checking `ref_Struct`
#1 [typeck_item_bodies] type-checking all item bodies
#2 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 8 previous errors

Some errors have detailed explanations: E0152, E0412, E0425, E0601.
For more information about an error, try `rustc --explain E0152`.
```

</p>
</details>

--- a/compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs
+++ b/compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs
@@ -2937,19 +2937,15 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             ObligationCauseCode::BinOp { rhs_span: Some(span), is_lit, .. } if *is_lit => span,
             _ => return,
         };
-        match (
-            trait_ref.skip_binder().self_ty().kind(),
-            trait_ref.skip_binder().substs.type_at(1).kind(),
-        ) {
-            (ty::Float(_), ty::Infer(InferTy::IntVar(_))) => {
-                err.span_suggestion_verbose(
-                    rhs_span.shrink_to_hi(),
-                    "consider using a floating-point literal by writing it with `.0`",
-                    ".0",
-                    Applicability::MaybeIncorrect,
-                );
-            }
-            _ => {}
+        if let ty::Float(_) = trait_ref.skip_binder().self_ty().kind()
+            && let ty::Infer(InferTy::IntVar(_)) = trait_ref.skip_binder().substs.type_at(1).kind()
+        {
+            err.span_suggestion_verbose(
+                rhs_span.shrink_to_hi(),
+                "consider using a floating-point literal by writing it with `.0`",
+                ".0",
+                Applicability::MaybeIncorrect,
+            );
         }
     }
 

--- a/src/test/ui/traits/issue-102989.rs
+++ b/src/test/ui/traits/issue-102989.rs
@@ -1,9 +1,6 @@
-//~ ERROR can't find crate for `profiler_builtins`
-// compile-flags: -Cinstrument-coverage
 // normalize-stderr-test "loaded from .*libcore-.*.rlib" -> "loaded from SYSROOT/libcore-*.rlib"
 
-#![no_core]
-#![feature(no_core, lang_items)]
+#![feature(lang_items)]
 #[lang="sized"]
 trait Sized { } //~ ERROR found duplicate lang item `sized`
 

--- a/src/test/ui/traits/issue-102989.rs
+++ b/src/test/ui/traits/issue-102989.rs
@@ -1,5 +1,6 @@
+//~ ERROR can't find crate for `profiler_builtins`
 // compile-flags: -Cinstrument-coverage
-//~^ ERROR can't find crate for `profiler_builtins`
+// normalize-stderr-test "loaded from .*libcore-.*.rlib" -> "loaded from SYSROOT/libcore-*.rlib"
 
 #![no_core]
 #![feature(no_core, lang_items)]

--- a/src/test/ui/traits/issue-102989.rs
+++ b/src/test/ui/traits/issue-102989.rs
@@ -1,0 +1,18 @@
+// compile-flags: -Cinstrument-coverage
+//~^ ERROR can't find crate for `profiler_builtins`
+
+#![no_core]
+#![feature(no_core, lang_items)]
+#[lang="sized"]
+trait Sized { } //~ ERROR found duplicate lang item `sized`
+
+fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
+    //~^ ERROR `self` parameter is only allowed in associated functions
+    //~| ERROR cannot find type `Struct` in this scope
+    //~| ERROR mismatched types
+    let x = x << 1;
+    //~^ ERROR the size for values of type `{integer}` cannot be known at compilation time
+    //~| ERROR cannot find value `x` in this scope
+}
+
+fn main() {}

--- a/src/test/ui/traits/issue-102989.stderr
+++ b/src/test/ui/traits/issue-102989.stderr
@@ -1,5 +1,5 @@
 error: `self` parameter is only allowed in associated functions
-  --> $DIR/issue-102989.rs:10:15
+  --> $DIR/issue-102989.rs:7:15
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |               ^^^^ not semantically valid as function parameter
@@ -7,50 +7,37 @@ LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    = note: associated functions are those in `impl` or `trait` definitions
 
 error[E0412]: cannot find type `Struct` in this scope
-  --> $DIR/issue-102989.rs:10:22
+  --> $DIR/issue-102989.rs:7:22
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |                      ^^^^^^ not found in this scope
 
 error[E0425]: cannot find value `x` in this scope
-  --> $DIR/issue-102989.rs:14:13
+  --> $DIR/issue-102989.rs:11:13
    |
 LL |     let x = x << 1;
    |             ^ help: a local variable with a similar name exists: `f`
 
-error: `profiler_builtins` crate (required by compiler options) is not compatible with crate attribute `#![no_core]`
-
-error[E0463]: can't find crate for `profiler_builtins`
-   |
-   = note: the compiler may have been built without the profiler runtime
-
 error[E0152]: found duplicate lang item `sized`
-  --> $DIR/issue-102989.rs:8:1
+  --> $DIR/issue-102989.rs:5:1
    |
 LL | trait Sized { }
    | ^^^^^^^^^^^
    |
-   = note: the lang item is first defined in crate `core`.
+   = note: the lang item is first defined in crate `core` (which `std` depends on)
    = note: first definition in `core` loaded from SYSROOT/libcore-*.rlib
    = note: second definition in the local crate (`issue_102989`)
 
-error: `#[panic_handler]` function required, but not found
-
-error: language item required, but not found: `eh_personality`
-   |
-   = note: this can occur when a binary crate with `#![no_std]` is compiled for a target where `eh_personality` is defined in the standard library
-   = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`
-
 error[E0277]: the size for values of type `{integer}` cannot be known at compilation time
-  --> $DIR/issue-102989.rs:14:15
+  --> $DIR/issue-102989.rs:11:15
    |
 LL |     let x = x << 1;
    |               ^^ doesn't have a size known at compile-time
    |
-   = help: the trait `core::marker::Sized` is not implemented for `{integer}`
+   = help: the trait `std::marker::Sized` is not implemented for `{integer}`
 
 error[E0308]: mismatched types
-  --> $DIR/issue-102989.rs:10:42
+  --> $DIR/issue-102989.rs:7:42
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |    ----------                            ^^^^ expected `&u32`, found `()`
@@ -58,7 +45,7 @@ LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |    implicitly returns `()` as its body has no tail or `return` expression
    |
 note: consider returning one of these bindings
-  --> $DIR/issue-102989.rs:10:30
+  --> $DIR/issue-102989.rs:7:30
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |                              ^
@@ -66,7 +53,7 @@ LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
 LL |     let x = x << 1;
    |         ^
 
-error: aborting due to 10 previous errors
+error: aborting due to 6 previous errors
 
-Some errors have detailed explanations: E0152, E0277, E0308, E0412, E0425, E0463.
+Some errors have detailed explanations: E0152, E0277, E0308, E0412, E0425.
 For more information about an error, try `rustc --explain E0152`.

--- a/src/test/ui/traits/issue-102989.stderr
+++ b/src/test/ui/traits/issue-102989.stderr
@@ -1,0 +1,72 @@
+error: `self` parameter is only allowed in associated functions
+  --> $DIR/issue-102989.rs:9:15
+   |
+LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
+   |               ^^^^ not semantically valid as function parameter
+   |
+   = note: associated functions are those in `impl` or `trait` definitions
+
+error[E0412]: cannot find type `Struct` in this scope
+  --> $DIR/issue-102989.rs:9:22
+   |
+LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
+   |                      ^^^^^^ not found in this scope
+
+error[E0425]: cannot find value `x` in this scope
+  --> $DIR/issue-102989.rs:13:13
+   |
+LL |     let x = x << 1;
+   |             ^ help: a local variable with a similar name exists: `f`
+
+error: `profiler_builtins` crate (required by compiler options) is not compatible with crate attribute `#![no_core]`
+
+error[E0463]: can't find crate for `profiler_builtins`
+   |
+   = note: the compiler may have been built without the profiler runtime
+
+error[E0152]: found duplicate lang item `sized`
+  --> $DIR/issue-102989.rs:7:1
+   |
+LL | trait Sized { }
+   | ^^^^^^^^^^^
+   |
+   = note: the lang item is first defined in crate `core`.
+   = note: first definition in `core` loaded from $BUILD_DIR/aarch64-apple-darwin/stage1/lib/rustlib/aarch64-apple-darwin/lib/libcore-500f4c12402b1108.rlib
+   = note: second definition in the local crate (`issue_102989`)
+
+error: `#[panic_handler]` function required, but not found
+
+error: language item required, but not found: `eh_personality`
+   |
+   = note: this can occur when a binary crate with `#![no_std]` is compiled for a target where `eh_personality` is defined in the standard library
+   = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`
+
+error[E0277]: the size for values of type `{integer}` cannot be known at compilation time
+  --> $DIR/issue-102989.rs:13:15
+   |
+LL |     let x = x << 1;
+   |               ^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `core::marker::Sized` is not implemented for `{integer}`
+
+error[E0308]: mismatched types
+  --> $DIR/issue-102989.rs:9:42
+   |
+LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
+   |    ----------                            ^^^^ expected `&u32`, found `()`
+   |    |
+   |    implicitly returns `()` as its body has no tail or `return` expression
+   |
+note: consider returning one of these bindings
+  --> $DIR/issue-102989.rs:9:30
+   |
+LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
+   |                              ^
+...
+LL |     let x = x << 1;
+   |         ^
+
+error: aborting due to 10 previous errors
+
+Some errors have detailed explanations: E0152, E0277, E0308, E0412, E0425, E0463.
+For more information about an error, try `rustc --explain E0152`.

--- a/src/test/ui/traits/issue-102989.stderr
+++ b/src/test/ui/traits/issue-102989.stderr
@@ -1,5 +1,5 @@
 error: `self` parameter is only allowed in associated functions
-  --> $DIR/issue-102989.rs:9:15
+  --> $DIR/issue-102989.rs:10:15
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |               ^^^^ not semantically valid as function parameter
@@ -7,13 +7,13 @@ LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    = note: associated functions are those in `impl` or `trait` definitions
 
 error[E0412]: cannot find type `Struct` in this scope
-  --> $DIR/issue-102989.rs:9:22
+  --> $DIR/issue-102989.rs:10:22
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |                      ^^^^^^ not found in this scope
 
 error[E0425]: cannot find value `x` in this scope
-  --> $DIR/issue-102989.rs:13:13
+  --> $DIR/issue-102989.rs:14:13
    |
 LL |     let x = x << 1;
    |             ^ help: a local variable with a similar name exists: `f`
@@ -25,13 +25,13 @@ error[E0463]: can't find crate for `profiler_builtins`
    = note: the compiler may have been built without the profiler runtime
 
 error[E0152]: found duplicate lang item `sized`
-  --> $DIR/issue-102989.rs:7:1
+  --> $DIR/issue-102989.rs:8:1
    |
 LL | trait Sized { }
    | ^^^^^^^^^^^
    |
    = note: the lang item is first defined in crate `core`.
-   = note: first definition in `core` loaded from $BUILD_DIR/aarch64-apple-darwin/stage1/lib/rustlib/aarch64-apple-darwin/lib/libcore-500f4c12402b1108.rlib
+   = note: first definition in `core` loaded from SYSROOT/libcore-*.rlib
    = note: second definition in the local crate (`issue_102989`)
 
 error: `#[panic_handler]` function required, but not found
@@ -42,7 +42,7 @@ error: language item required, but not found: `eh_personality`
    = help: you may be able to compile for a target that doesn't need `eh_personality`, specify a target with `--target` or in `.cargo/config`
 
 error[E0277]: the size for values of type `{integer}` cannot be known at compilation time
-  --> $DIR/issue-102989.rs:13:15
+  --> $DIR/issue-102989.rs:14:15
    |
 LL |     let x = x << 1;
    |               ^^ doesn't have a size known at compile-time
@@ -50,7 +50,7 @@ LL |     let x = x << 1;
    = help: the trait `core::marker::Sized` is not implemented for `{integer}`
 
 error[E0308]: mismatched types
-  --> $DIR/issue-102989.rs:9:42
+  --> $DIR/issue-102989.rs:10:42
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |    ----------                            ^^^^ expected `&u32`, found `()`
@@ -58,7 +58,7 @@ LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |    implicitly returns `()` as its body has no tail or `return` expression
    |
 note: consider returning one of these bindings
-  --> $DIR/issue-102989.rs:9:30
+  --> $DIR/issue-102989.rs:10:30
    |
 LL | fn ref_Struct(self: &Struct, f: &u32) -> &u32 {
    |                              ^

